tests/nativeapp/test_package_scripts.py

```diff
@@ -175,17 +197,26 @@ def test_package_scripts_w_warehouse_access_exception(
     mock_conn,
     mock_execute_query,
     project_definition_files,
+    mock_cursor,
 ):
+    side_effects = [
+        mock_cursor([{"CURRENT_WAREHOUSE()": "old_wh"}], []),
+        ProgrammingError(
+            msg="Object does not exist, or operation cannot be performed.",
+            errno=DOES_NOT_EXIST_OR_CANNOT_BE_PERFORMED,
+        ),
+    ]
+
     mock_conn.return_value = MockConnectionCtx()
-    mock_execute_query.side_effect = ProgrammingError(
-        msg="Object does not exist, or operation cannot be performed.",
-        errno=DOES_NOT_EXIST_OR_CANNOT_BE_PERFORMED,
-    )
+    mock_execute_query.side_effect = side_effects
 
     working_dir: Path = project_definition_files[0].parent
     native_app_manager = _get_na_manager(str(working_dir))
 
     with pytest.raises(ProgrammingError) as err:
         native_app_manager._apply_package_scripts()  # noqa: SLF001
 
-    assert "Please grant usage privilege on warehouse to this role." in err.value.msg
+    assert (
+        "Could not use warehouse MockWarehouse. Object does not exist, or operation cannot be performed."
+        in err.value.msg
+    )
```

Comment:
This error message is now different because instead of calling `use warehouse` directly which uses our `generic_error_handler`, it calls the method `self.use` provided by the CLI which has its own error handling.